src/compiler/checker.ts

```diff
@@ -11113,6 +11113,7 @@ namespace ts {
             // names and consistency of modifiers are verified when we check local symbol
             let isExportSymbolInsideModule = symbol.parent && symbol.parent.flags & SymbolFlags.Module;
             let duplicateFunctionDeclaration = false;
+            let multipleDefaultModifiers = false;

```

Comment:
Actually, don't do checking here at all. The user is going to get a 'Duplicate function implementation' error anyway, which is really the main problem they have to address.

![image](https://cloud.githubusercontent.com/assets/972891/10353428/5ffaf64a-6d0b-11e5-93d6-3e22ec933350.png)